### PR TITLE
Improve provision scripts for aws

### PR DESCRIPTION
**Pull request by Jeffwan**
Get some feedbacks from AWS users and make some improvements on kfctl.sh

1. Make sure TLS and authentication can work with custom configuration
2. Improve storage PV provision parts. Add static and dynamic for Lustre
3. Use cluster_config to provision cluster rather than arguments from kfctl which will make kfctl looks cleaner. 
4. Add private access and control panel logs support.
5. Fix several bugs due to different environment and setups. Didn't see extra issues from users.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2983)

### CODE REVIEW

File: scripts/kfctl.sh [hunk 1]
```diff
@@ -657,8 +634,6 @@ main() {
       else
         echo "namespace ${K8S_NAMESPACE} successfully deleted."
       fi
-      # double confirm resources not in kubeflow namespace are deleted
```
Comment (Jeffwan) [inline scripts/kfctl.sh]: In last commit https://github.com/kubeflow/kubeflow/pull/2823/files#diff-8a6f2d237f052b07ca594c20830ed9d2R661,  add this line to make sure resources are cleaned. Notice this is unneeded.

File: scripts/kfctl.sh [hunk 2]
```diff
@@ -323,33 +316,17 @@ parseArgs() {
         shift
         AZ_NODE_SIZE=$1
         ;;
-      --clusterName)
```
Comment (Jeffwan) [inline scripts/kfctl.sh]: Only reserve required 3 commands. Move rest of them to cluster_config.yaml

File: scripts/kfctl.sh [hunk 3]
```diff
@@ -110,26 +110,19 @@ createEnv() {
       ;;
     aws)
       export KUBEFLOW_PLATFORM=aws
-      INPUT+=('CLUSTER_NAME=$CLUSTER_NAME\n'
```
Comment (Jeffwan) [inline scripts/kfctl.sh]: Simplify envs.

**Comment by Jeffwan** [comment]
@jlewi  Hi Jeremy, could you help take a look at this PR? I get some feedbacks from users and trying to  simplify setups. Most of the changes are aws specific.

**Comment by jlewi** [comment]
/ok-to-test
/lgtm
/approve

@Jeffwan Sorry for the slow reply.  If you create an OWNERs file in the AWS specific directory then you can list yourself and other folks as approvers and it will be easier to get those changes reviewed.

**Pull request merged**